modules/openxr/extensions/openxr_composition_layer_extension.cpp

```diff
@@ -256,49 +266,147 @@ void OpenXRViewportCompositionLayerProvider::set_use_android_surface(bool p_use_
 #endif
 }
 
-#ifdef ANDROID_ENABLED
-void OpenXRViewportCompositionLayerProvider::create_android_surface() {
-	ERR_FAIL_COND(android_surface.swapchain != XR_NULL_HANDLE || android_surface.surface.is_valid());
-	ERR_FAIL_COND(!openxr_api || !openxr_api->is_running());
-
-	void *next_pointer = nullptr;
-	for (OpenXRExtensionWrapper *wrapper : openxr_api->get_registered_extension_wrappers()) {
-		void *np = wrapper->set_android_surface_swapchain_create_info_and_get_next_pointer(extension_property_values, next_pointer);
-		if (np != nullptr) {
-			next_pointer = np;
+void OpenXRCompositionLayerExtension::CompositionLayer::set_alpha_blend(bool p_alpha_blend) {
+	if (alpha_blend != p_alpha_blend) {
+		alpha_blend = p_alpha_blend;
+		if (alpha_blend) {
+			composition_layer.layerFlags |= XR_COMPOSITION_LAYER_BLEND_TEXTURE_SOURCE_ALPHA_BIT;
+		} else {
+			composition_layer.layerFlags &= ~XR_COMPOSITION_LAYER_BLEND_TEXTURE_SOURCE_ALPHA_BIT;
 		}
 	}
+}
 
-	// The XR_FB_android_surface_swapchain_create extension mandates that format, sampleCount,
-	// faceCount, arraySize, and mipCount must be zero.
-	XrSwapchainCreateInfo info = {
-		XR_TYPE_SWAPCHAIN_CREATE_INFO, // type
-		next_pointer, // next
-		0, // createFlags
-		XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_COLOR_ATTACHMENT_BIT | XR_SWAPCHAIN_USAGE_MUTABLE_FORMAT_BIT, // usageFlags
-		0, // format
-		0, // sampleCount
-		(uint32_t)swapchain_size.x, // width
-		(uint32_t)swapchain_size.y, // height
-		0, // faceCount
-		0, // arraySize
-		0, // mipCount
+void OpenXRCompositionLayerExtension::CompositionLayer::set_transform(const Transform3D &p_transform) {
+	Transform3D reference_frame = XRServer::get_singleton()->get_reference_frame();
+	Transform3D transform = reference_frame.inverse() * p_transform;
+	Quaternion quat(transform.basis.orthonormalized());
+
+	XrPosef pose = {
+		{ (float)quat.x, (float)quat.y, (float)quat.z, (float)quat.w },
+		{ (float)transform.origin.x, (float)transform.origin.y, (float)transform.origin.z }
 	};
 
-	jobject surface;
-	composition_layer_extension->create_android_surface_swapchain(&info, &android_surface.swapchain, &surface);
+	switch (composition_layer.type) {
```

Comment:
Similar to my previous remark, this needs to become more generic. If we're going down the path of using `OpenXRStructureBase` as I suggested, we could simply create a `set_pose` (or probably better a `set_transform` so it can be exposed to GDExtensions) on `OpenXRCompositionLayerBaseHeader` as a virtual method that we implement on each subclass.